### ISSUE TITLE
Brand name matches correctly ignores case, but not hiragana/katakana

### ISSUE THREAD

**Issue by ArenaL5** [openstreetmap/iD#7565]
When adding a point and typing a brand name, the search tool correctly ignores case: creating a point and typing a name will suggest matches in osmlab/name-suggestion-index with any capitalization:

![osm1](https://user-images.githubusercontent.com/37388405/80920990-a9ef2900-8d73-11ea-8bd1-5a43c424f522.png)

This works for any writing system with uppercase and lowercase letters. Japanese doesn't have that distinction, but does have hiragana/katakana. Every possible sound written in [hiragana](https://en.wikipedia.org/wiki/Hiragana) is paired with a sound in [katakana](https://en.wikipedia.org/wiki/Katakana).

As explained in [osmlab/name-suggestion-index#3806](https://github.com/osmlab/name-suggestion-index/pull/3806#issuecomment-623029149), the name search should ignore hiragana/katakana distinction like it ignores letter case, but that does not happen (シマムラ should suggest しまむら):

![osm2](https://user-images.githubusercontent.com/37388405/80920996-b07da080-8d73-11ea-9a1c-292565bad5b9.png)
![osm3](https://user-images.githubusercontent.com/37388405/80920998-b5425480-8d73-11ea-9d80-4d45fbd02b2d.png)

Please forgive me if this is too wordy, I'm finding it difficult to keep it short. I'm using Firefox 75.0 64-bit under Ubuntu 19.10. My addons are uBlock Origin 1.26.2 and tumpio's Request Control 1.15.2.

**Comment by 1ec5** [comment]
If I understand correctly, the issue is that iD only finds name-suggestion-index presets based on the `name`, not any other name tags or `matchNames`. This issue affects all writing systems. For example, searching for “Kentucky Fried Chicken” doesn’t return the preset that “KFC” does, even though [its entry in NSI](https://github.com/osmlab/name-suggestion-index/blob/70fdc02b0f862fc5d07dd71937275bc8dd1e9089/brands/amenity/fast_food.json#L1498-L1509) is tagged with `alt_name`.

**Comment by ArenaL5** [comment]
No, that's a different issue. `matchNames` work, it's just that "Kentucky Fried Chicken" is an `alt_name` but not a `matchName`.

What I meant is that the search engine should treat hiragana/katakana like uppercase/lowercase. If AKIO/akio/Akio/akIo/aKIO all suggest the same result, シマムラ/しまむら/しまムラ/シマむら should too.

The general idea is to treat characters [in every cell of this table](https://en.wikipedia.org/wiki/Kunrei-shiki_romanization#Kunrei-shiki%20spellings%20of%20kana) as one and the same. But some characters are not on the table, so you might want to look at other implementations.

[Here's an example in Python 2](https://pypi.org/project/jcconv/0.1.2/) (the ones that I found in JavaScript are ridiculously complex). A newer version in Python 3 can be found at LIQRGV/jcconv or https://pypi.org/project/jcconv3/. You could apply ~`half2hira`~, `kata2hira` and `wide2half` to both the string the user is typing, and the entries in the NSI. (If you prefer to create your own table of mappings, they're all unambiguous codepoint to codepoint mappings, so it should run as fast as it does now. For example, ﾅ and ナ always map to な)

EDITED: `half2hira` is more complex than I thought, and probably not worth it. `kata2hira` (katakana to hiragana) and `wide2half` (wide Latin letters to normal Latin letters) _are_ codepoint to codepoint mappings.

**Comment by 1ec5** [comment]
Yes, I’m aware of the possibility of automatically transliterating between Japanese writing systems. However, iD is written in JavaScript and deployed as part of a webpage, which makes bundle size a key performance consideration. Language-specific dependencies can be problematic. An alternative would be to manually specify the kana representations of each brand in `matchNames`. `matchNames` won’t appear in tagged OSM elements, so OSM’s guidance against tagging transliterations isn’t a problem.

**Comment by ArenaL5** [comment]
I understand. It's clearly not a trivial problem to solve as I hoped.

It _might_ be possible to leverage functions from [`Intl.Collator()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator), as those are already built in every modern browser. But I'd need to do several experiments to be certain.

I'll use `matchNames` for now. Thank you for your time.
